phylo2vec/src/utils.rs

```diff
@@ -0,0 +1,21 @@
+use rand::Rng;
+
+pub fn sample_ordered(n_leaves: usize) -> Vec<usize> {
+    let mut v: Vec<usize> = Vec::with_capacity(n_leaves - 1);
+    let mut rng = rand::thread_rng();
+
+    for i in 0..(n_leaves - 1) {
+        v.push(rng.gen_range(0..(i + 1)));
+    }
+    v
+}
+
+pub fn sample_unordered(n_leaves: usize) -> Vec<usize> {
+    let mut v: Vec<usize> = Vec::with_capacity(n_leaves - 1);
+    let mut rng = rand::thread_rng();
+
+    for i in 0..(n_leaves - 1) {
+        v.push(rng.gen_range(0..(2 * i + 1)));
+    }
+    v
+}
```

Comment:
```suggestion

#[derive(Default)]
pub enum SampleOrdering {
    #[default]
    Ordered,
    NotOrdered
}

pub fn sample(n_leaves: usize, ordering: SampleOrdering) -> Vec<usize> {
    let mut v: Vec<usize> = Vec::with_capacity(n_leaves - 1);
    let mut rng = rand::thread_rng();

    match ordering {
        SampleOrdering::Ordered => {
            for i in 0..(n_leaves - 1) {
                v.push(rng.gen_range(0..(i + 1)));
            }
        }
        SampleOrdering::NotOrdered => {
            for i in 0..(n_leaves - 1) {
                v.push(rng.gen_range(0..(2 * i + 1)));
            }
        }
    }

    v
}

```
In Rust, to give default values, you can use the [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) trait. To tackle this, you want to create some sort of enum for this since the values are really limited to options.